File: Python-implementation/Levenshtein.py
def calculate(str1, str2):
    dp = [[0] * (len(str2) + 1) for _ in range(len(str1) + 1)]

    for i in range(len(str1) + 1):
        dp[i][0] = i
    for j in range(len(str2) + 1):
        dp[0][j] = j

    for i in range(1, len(str1) + 1):
        for j in range(1, len(str2) + 1):
            if str1[i - 1] == str2[j - 1]:
                dp[i][j] = dp[i - 1][j - 1]
            else:
                dp[i][j] = min(dp[i - 1][j - 1], dp[i][j - 1], dp[i - 1][j]) + 1

    return dp[-1][-1]

def find_closest_word(target, vocabulary):
    closest_word = ""
    min_distance = float('inf')

    for word in vocabulary:
        distance = calculate(target, word)
        if distance < min_distance:
            min_distance = distance
            closest_word = word

    return closest_word

if __name__ == "__main__":
    target = "kitten"
    vocabulary = ["sitting", "kitchen", "mittens", "biting", "mitten"]
    closest_word = find_closest_word(target, vocabulary)
    
    print("Closest word to '{}' is: {}".format(target, closest_word))

File: Java-implementation/src/Levenshtein.java
import java.util.*;

public class Levenshtein {
    
    public static int calculate(String str1, String str2) {
        int[][] dp = new int[str1.length() + 1][str2.length() + 1];
        
        for (int i = 0; i <= str1.length(); i++) {
            dp[i][0] = i;
        }
        for (int j = 0; j <= str2.length(); j++) {
            dp[0][j] = j;
        }
        
        for (int i = 1; i <= str1.length(); i++) {
            for (int j = 1; j <= str2.length(); j++) {
                if (str1.charAt(i - 1) == str2.charAt(j - 1)) {
                    dp[i][j] = dp[i - 1][j - 1];
                } else {
                    dp[i][j] = Math.min(dp[i - 1][j - 1], Math.min(dp[i][j - 1], dp[i - 1][j])) + 1;
                }
            }
        }
        
        return dp[str1.length()][str2.length()];
    }
    
    public static String findClosestWord(String target, List<String> vocabulary) {
        String closestWord = "";
        int minDistance = Integer.MAX_VALUE;
        
        for (String word : vocabulary) {
            int distance = calculate(target, word);
            if (distance < minDistance) {
                minDistance = distance;
                closestWord = word;
            }
        }
        
        return closestWord;
    }

    public static void main(String[] args) {
        String target = "kitten";
        List<String> vocabulary = Arrays.asList("sitting", "kitchen", "mittens", "biting", "mitten");
        String closestWord = findClosestWord(target, vocabulary);
        
        System.out.println("Closest word to '" + target + "' is: " + closestWord);
    }
}